src/ol/array.js

```diff
@@ -127,3 +164,147 @@ ol.array.reverseSubArray = function(arr, begin, end) {
     --end;
   }
 };
+
+
+/**
+ * @param {Array.<*>} arr Array.
+ * @return {!Array<?>} Flattened Array.
+ */
+ol.array.flatten = function(arr) {
+  var data = arr.reduce(function(flattened, value) {
+    if (goog.isArray(value)) {
+      return flattened.concat(ol.array.flatten(value));
+    } else {
+      return flattened.concat(value);
+    }
+  }, []);
+  return data;
+};
+
+
+/**
+ * @param {Array<VALUE>} arr  The array to modify.
+ * @param {Array<VALUE>|VALUE} data The elements or arrays of elements
+ *     to add to arr.
+ * @template VALUE
+ */
+ol.array.extend = function(arr, data) {
+  var i;
+  var extension = goog.isArrayLike(data) ? data : [data];
+  var length = extension.length
+  for (i = 0; i < length; i++) {
+    arr[arr.length] = extension[i];
+  }
+}
+
+
+/**
+ * @param {Array<VALUE>} arr  The array to modify.
+ * @param {VALUE} obj The element to remove.
+ * @template VALUE
+ * @return {boolean} If the element was removed.
+ */
+ol.array.remove = function(arr, obj) {
+  var i = arr.indexOf(obj);
+  var found = i > -1;
+  if (found) {
+    arr.splice(i, 1);
+  }
+  return found;
+}
+
+
+/**
+ * @param {Array<VALUE>} arr  The array to search in.
+ * @param {function(VALUE, number, ?) : boolean} func The function to compare.
+ * @template VALUE
+ * @return {VALUE} The element found.
+ */
+ol.array.find = function(arr, func) {
+  var length = arr.length >>> 0;
+  var value;
+
+  for (var i = 0; i < length; i++) {
+    value = arr[i];
+    if (func(value, i, arr)) {
+      return value;
+    }
+  }
+  return null;
+}
+
+
+/**
+* @param {Array|NodeList|Arguments|{length: number}} arr1 The first array to compare.
+* @param {Array|NodeList|Arguments|{length: number}} arr2 The second array to compare.
+* @return {boolean} Whether the two arrays are equal.
+ */
+ol.array.equals = function(arr1, arr2) {
+  if (!goog.isArrayLike(arr1) || !goog.isArrayLike(arr2) || arr1.length !== arr2.length) {
+    return false;
+  }
+  var length = arr1.length;
+  var equalsFn = function(a, b) {
+    return a === b;
+  };
+  for (var i = 0; i < length; i++) {
+    if (!equalsFn(arr1[i], arr2[i])) {
+      return false;
+    }
+  }
+  return true;
+}
+
+
+/**
+* @param {Array<*>} arr The array to sort (modifies original).
+* @param {Function} compareFnc Comparison function.
+ */
+ol.array.stableSort = function(arr, compareFnc) {
+  var length = arr.length;
+  var tmp = Array(arr.length);
+  var i;
+  for (i = 0; i < length; i++) {
+    tmp[i] = {index: i, value: arr[i]};
+  }
+  var compare = compareFnc || ol.array.numberSafeCompareFunction;
```

Comment:
Since the `compareFnc` argument is not optional, you can use `compareFnc` directly below. No need to fall back to `ol.array.numberSafeCompareFunction`.